dashboard/app/models/census/census_summary.rb

```diff
@@ -206,24 +228,74 @@ def self.summarize_state_data(school, school_year, high_school, state_years_with
   # 8	teaches_cs from 2 years ago
   # 9 nil
   #
-  def self.compute_teaches_cs(has_ap_data, has_ib_data, submissions_summary, state_summary, previous_years_results)
-    if has_ap_data || has_ib_data
-      'YES'
-    elsif submissions_summary[:consistency][:teacher_or_admin]
-      submissions_summary[:consistency][:teacher_or_admin]
-    elsif state_summary
-      state_summary
-    elsif submissions_summary[:consistency][:not_teacher_or_admin]
-      submissions_summary[:consistency][:not_teacher_or_admin]
-    elsif submissions_summary[:has_inconsistent_surveys]
-      'MAYBE'
-    elsif map_historical_teaches_cs previous_years_results.first
-      map_historical_teaches_cs previous_years_results.first
-    elsif map_historical_teaches_cs previous_years_results.second
-      map_historical_teaches_cs previous_years_results.second
-    else
-      nil
-    end
+  def self.compute_teaches_cs(has_ap_data, has_ib_data, submissions_summary, state_summary, previous_years_results, audit)
+    result = nil
+    details = []
+
+    result = conditional_result('offers_ap', has_ap_data, 'YES', true, result, details)
+    result = conditional_result('offers_ib', has_ib_data, 'YES', true, result, details)
+    result = conditional_result(
+      'consistent_teacher_surveys',
+      submissions_summary[:consistency][:teacher_or_admin],
+      submissions_summary[:consistency][:teacher_or_admin],
+      submissions_summary[:counts][:teacher_or_admin],
+      result,
+      details
+    )
+
+    state_value = {
+      'NO' => false,
+      'YES' => true,
+      nil => nil
+    }[state_summary]
+
+    result = conditional_result('state_offering', state_summary, state_summary, state_value, result, details)
+
+    result = conditional_result(
```

Comment:
Maybe a comment somewhere explaining the logic of chaining these function results together?

Reply:
Added.